docs/API_POLICY.md

```diff
@@ -0,0 +1,39 @@
+# Hub API Policy
+This document proposes a policy regarding making API updates in this repo.
+
+There are 2 types of APIs:
+## 1. Versioned APIs: 
+- These are the APIs which are exposed for external users to use and integrate with their applications.
+- These will have support provided by the Hub team.
+- These are the APIs which return hub resource data.
+- These are versioned, the version of the APIs will be prefixed to the API route. 
+Eg. resource API - /v1/resources
+- Hub Release Version has nothing to do with API version. API version will be incremented only when there is a breaking change. Making a hub major release not necessarily increment API version too.
```

Comment:
> Making a hub major release not necessarily increment API version too.

IMHO if there is a major release in API side then I guess we should be incrementing API version too as it will help users to differentiate between older and newer version. WDYT?

Reply:
Hub is like a service that will keep evolving so the hub version doesn't really matter to the user as compared to any CLI version where the user needs to download the binary and run it.
Hub may have major changes in UI which might not have impact on API still incrementing API version would be unnecessary.
due to which hub release version and API version are independent. API version will increment only if there is a breaking change.

Reply:
So if I'm not wrong do you mean Hub release version and API release version might not go in sync in future ??

Reply:
yes